src/instance.rs

```diff
@@ -159,9 +161,10 @@ impl fmt::Display for Token {
 pub struct ChorusUser {
     pub belongs_to: Shared<Instance>,
     pub token: String,
+    pub mfa_token: Option<MfaToken>,
     pub limits: Option<HashMap<LimitType, Limit>>,
     pub settings: Shared<UserSettings>,
-    pub object: Shared<User>,
+    pub object: Option<Shared<User>>,
```

Comment:
Hmm, the necessity for this change makes me question the architecture I came up with for `ChorusUser` a little. I'd like to avoid wrapping the `object` property of `ChorusUser` in an `Option<T>`, since `object` is accessed a lot, and the addition of an `Option<T>`, in my opinion, drastically makes the API usage of `ChorusUser` more cluttered, as can be seen in `examples/login.rs`:

```diff
-     dbg!(&user.object.read().unwrap().username);
+     dbg!(&user.object.unwrap().as_ref().read().unwrap().username);
```

Perhaps there should be something like a `ChorusUserTBA` (Chorus User To-Be-Authenticated), which we use internally pre-auth, and which yields a `ChorusUser` object post-auth. This way, we could be sure that the user has been authenticated, without requiring the `Option<T>`. I definitely see that calling `User::default()` pre-auth is a little silly, and that we should have a way to be sure that the user has been authenticated (at least once).

This is out of scope for this PR though, and likely worthy of its' own issue.
Do you @kozabrada123 and @xystrive understand what I mean? Does my suggestion make sense? (Genuine question). Would be happy to receive feedback on this :)

Reply:
Well, to me this is a question of if we need something like `ChorusUserTBA`

Do we _need_ an unauthenticated `ChorusUser`? In the past we've put routes pre-authentication mostly in `Instance`, so is there anything that would require a partial `ChorusUser`?

`send_request` and `deserialize_response` both need a `ChorusUser` for its `limits` field (for ratelimited requests), I suppose we could create an unauthenticated user object and then maybe a trait to allow passing both? Or just make the functions take an `&mut Option<HashMap<LimitType, Limit>>`

Reply:
> Hmm, the necessity for this change makes me question the architecture I came up with for `ChorusUser` a little. I'd like to avoid wrapping the `object` property of `ChorusUser` in an `Option<T>`, since `object` is accessed a lot, and the addition of an `Option<T>`, in my opinion, drastically makes the API usage of `ChorusUser` more cluttered, as can be seen in `examples/login.rs`:
> 
> ```diff
> -     dbg!(&user.object.read().unwrap().username);
> +     dbg!(&user.object.unwrap().as_ref().read().unwrap().username);
> ```
> 
> Perhaps there should be something like a `ChorusUserTBA` (Chorus User To-Be-Authenticated), which we use internally pre-auth, and which yields a `ChorusUser` object post-auth. This way, we could be sure that the user has been authenticated, without requiring the `Option<T>`. I definitely see that calling `User::default()` pre-auth is a little silly, and that we should have a way to be sure that the user has been authenticated (at least once).
> 
> This is out of scope for this PR though, and likely worthy of its' own issue. Do you @kozabrada123 and @xystrive understand what I mean? Does my suggestion make sense? (Genuine question). Would be happy to receive feedback on this :)

I agree, wrapping the `object` field of `ChorusUser` with `Option` makes it a bit messy.
In my opinion I think ChorusUser should only represent an authenticated user. I am open to give my input about this design consideration more thoroughly in the appropriate place.

Reply:
Thanks for the feedback, both of you :) 
It seems like regardless of which concrete solution we end up going with, `Option<ChorusUser>` is not going to stay. However, I think you can keep `Option<ChorusUser>` for now, since it emphasizes the problem we have. Tidying it up to be a nicer solution will be a separate issue. :)